zen/cursor.c

```diff
@@ -0,0 +1,107 @@
+#include "zen/cursor.h"
+
+#include <drm/drm_fourcc.h>
+#include <linux/input.h>
+#include <wlr/render/wlr_renderer.h>
+#include <wlr/types/wlr_seat.h>
+#include <wlr/xcursor.h>
+
+#include "zen-common.h"
+#include "zen/scene/screen-layout.h"
+#include "zen/server.h"
+
+static void
+zn_cursor_handle_destroy_screen(struct wl_listener* listener, void* data)
+{
+  UNUSED(data);
+  struct zn_cursor* self = zn_container_of(listener, self, new_screen_listener);
+  struct zn_server* server = zn_server_get_singleton();
+  struct zn_screen_layout* screen_layout = server->scene->screen_layout;
+
+  wl_list_remove(&self->destroy_screen_listener.link);
```

Comment:
```suggestion
  wl_list_remove(&self->destroy_screen_listener.link);
  wl_list_init(&self->destroy_screen_listener.link);
```
You need to init the list. Otherwise `wl_list_remove(&self->destroy_screen_listener.link);` is executed again in the destructor, and it's not safe.